ecosystem/typescript/sdk/src/utils/memoize-decorator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Credits to https://github.com/darrylhodgins/typescript-memoize
+ */
+
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+
+interface MemoizeArgs {
+  expiring?: number;
+  hashFunction?: boolean | ((...args: any[]) => any);
+  tags?: string[];
+}
+
+export function Memoize(args?: MemoizeArgs | MemoizeArgs["hashFunction"]) {
```

Comment:
Wow this is sorta weird, you can declare a type using the field of another type?

Reply:
This function is a decorator factory. What problems do you see?

Reply:
Nothing, just never seen this kind of thing in a language before.